src/misc/fs_utils_win32.cpp

```diff
@@ -66,18 +66,23 @@ constexpr uint8_t WindowsAttributesMask = 0x3f;
 FILE* local_drive_create_file(const std_fs::path& path,
                               const FatAttributeFlags attributes)
 {
-	// NOTE: If changing this implementation, test if 'Crystal Caves' game
-	// can save a game to the same slot it was loaded from. Previous
-	// implementation, utilizing 'CreateFileW', did not work in such case,
-	// game was producing FILE ERROR - see the issue:
-	// https://github.com/dosbox-staging/dosbox-staging/issues/3262
-
-	FILE* file_pointer = fopen(path.string().c_str(), "wb+");
-	if (file_pointer) {
-		if (!SetFileAttributesW(path.c_str(), attributes._data)) {
-			fclose(file_pointer);
-			return nullptr;
-		}
+	FILE* file_pointer          = nullptr;
+	const auto win32_attributes = (attributes._data != 0)
+	                                    ? static_cast<DWORD>(attributes._data)
+	                                    : FILE_ATTRIBUTE_NORMAL;
+	const auto win32_handle     = CreateFileW(path.c_str(),
```

Comment:
picky: maybe insert a new line above this and reformat?

Reply:
Formatted